plutus-tx/src/PlutusTx/Blueprint/Definition.hs

```diff
@@ -36,11 +47,47 @@ import PlutusTx.Blueprint.Schema (Schema (..))
 import PlutusTx.Builtins.Internal (BuiltinByteString, BuiltinData, BuiltinList, BuiltinString,
                                    BuiltinUnit)
 
--- For more context see the note ["Unrolling" types]
+-- | A schema definition of a type @t@ with a list of referenced types @ts@.
+data Definition t ts = MkDefinition DefinitionId (Schema ts)
+  deriving stock (Show)
+
+-- | A registry of schema definitions.
+data Definitions (ts :: [Type]) where
+  NoDefinitions :: Definitions '[]
+  DCons :: SomeDefinition -> Definitions ts -> Definitions (t ': ts)
```

Comment:
This seems wrong, it doesn't mention `t`!! You can always quantify over the variables in this constructor also, you don't need `SomeDefinition` AFAICT.

i.e.
```
DCons :: forall ts' . Definition t ts' -> Definitions ts -> Definitions (t ': ts)
```

Reply:
TBH, I am also confused by the fact that `ts'` would be unused there...